### PR TITLE
Add viz and jaeger list commands

### PR DESCRIPTION
**Pull request by adleong**
Pods can only participate in tracing if they have been injected by the jaeger-injector.  Similarly, pods may only be tapped if they have been injected by the tap-injector.  Since pods which were already running when the injector starts will not be injected until those pods are restarted, it can be difficult to know which pods in your cluster will participate in tracing or be valid tap targets.

We add the command `linkerd viz list` to list meshed pods and indicate which can be tapped, which need to be restarted before they can be tapped, and which have tap disabled.

```console
> linkerd viz list -A
Pods with tap enabled:
	* collector-7f585dc68-z8vc8.linkerd-jaeger
	* jaeger-69fc767648-mxtc4.linkerd-jaeger
	* jaeger-injector-67fbccc487-sjh4c.linkerd-jaeger
	* grafana-84c9b569b9-27qsj.linkerd-viz
	* metrics-api-6c956b4b58-5xvz8.linkerd-viz
	* prometheus-7fdb866467-s4q5m.linkerd-viz
	* tap-768b5ddc6c-hdfb2.linkerd-viz
	* tap-injector-ff446c479-4wtsm.linkerd-viz
	* web-5f79854c4f-lpv5l.linkerd-viz
Pods missing tap configuration (restart these pods to enable tap):
	* linkerd-gateway-777b7cb9bf-7fr2n.linkerd-multicluster
	* linkerd-controller-6864cf5f8c-bxp7l.linkerd
	* linkerd-destination-67499b8df7-fqqb9.linkerd
	* linkerd-identity-7c577f7454-c2v7r.linkerd
	* linkerd-proxy-injector-c7844b9f6-hwbsm.linkerd
	* linkerd-sp-validator-f4c8cc6d6-658tb.linkerd
``` 

Similarly, we add the command `linkerd jaeger list` to list meshed pods and indicate which will participate in tracing.

```console
> linkerd jaeger list -A
Pods with tracing enabled:
	* grafana-84c9b569b9-27qsj.linkerd-viz
	* metrics-api-6c956b4b58-5xvz8.linkerd-viz
	* prometheus-7fdb866467-s4q5m.linkerd-viz
	* tap-768b5ddc6c-hdfb2.linkerd-viz
	* tap-injector-ff446c479-4wtsm.linkerd-viz
	* web-5f79854c4f-lpv5l.linkerd-viz
Pods missing tracing configuration (restart these pods to enable tracing):
	* collector-7f585dc68-z8vc8.linkerd-jaeger
	* jaeger-69fc767648-mxtc4.linkerd-jaeger
	* jaeger-injector-67fbccc487-sjh4c.linkerd-jaeger
	* linkerd-gateway-777b7cb9bf-7fr2n.linkerd-multicluster
	* linkerd-controller-6864cf5f8c-bxp7l.linkerd
	* linkerd-destination-67499b8df7-fqqb9.linkerd
	* linkerd-identity-7c577f7454-c2v7r.linkerd
	* linkerd-proxy-injector-c7844b9f6-hwbsm.linkerd
	* linkerd-sp-validator-f4c8cc6d6-658tb.linkerd
```

This commands list pods in the context's default namespcae by default, but can be configured with the usual `-n` and `-A` flags.

This replaces the jaeger extension's data plane check which gave a warning if there were pods with tracing.  That check has been removed here.

### CODE REVIEW

File: jaeger/cmd/list.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	namespace     string
+	allNamespaces bool
+}
+
+func newCmdList() *cobra.Command {
+	var options listOptions
+
+	cmd := &cobra.Command{
+		Use:   "list [flags]",
+		Short: "Lists which pods can be tapped",
```
Comment (dadjeibaah) [inline jaeger/cmd/list.go:26]: Should this say `Lists which pods can be traced`?

**Review by dadjeibaah — COMMENTED** [review]
The pod names formatting in the list command looks a little different compared to how they looked in the check command output.
On this branch it looks like we format the names in `<pod-name>.<namespace>`
```bash
	* collector-69cc44dfbc-sbkxp.linkerd-jaeger
	* jaeger-6f98d5c979-kwqft.linkerd-jaeger
	* jaeger-injector-69985d45db-rlz4v.linkerd-jaeger
```

But in the check output we do:
```bash
	* linkerd-jaeger/collector-69cc44dfbc-sbkxp
	* linkerd-jaeger/jaeger-6f98d5c979-kwqft
	* linkerd-jaeger/jaeger-injector-69985d45db-rlz4v
```

Do we want to keep the same format as before or are we looking to change it?

**Review by kleimkuhler — APPROVED** [review]
Looks good and works well:

```
❯ ./target/release/linkerd2-cli-git-1ba566af-linux-amd64 viz list -A
Pods with tap enabled:
        * linkerd-jaeger/jaeger-69fc767648-ljvvl
        * linkerd-jaeger/jaeger-injector-7f4b5df5b4-n4lhr
        * linkerd-jaeger/collector-7f585dc68-7sdbh
Pods missing tap configuration (restart these pods to enable tap):
        * linkerd/linkerd-identity-7ccdfb455c-fbhsj
        * linkerd/linkerd-sp-validator-57878ff5cc-rshng
        * linkerd/linkerd-controller-74bbd789c7-59v75
        * linkerd/linkerd-destination-6f8977f44f-72bhg
        * linkerd/linkerd-proxy-injector-7664989ff9-gghrv
        * linkerd-viz/tap-759698d4c8-vrzmc
        * linkerd-viz/tap-injector-865cfdfb7d-89np8
        * linkerd-viz/metrics-api-9b9b4d79f-b42qp
        * linkerd-viz/grafana-5b8bc7dd5d-dtkqg
        * linkerd-viz/web-957c75976-8v7cx
        * linkerd-viz/prometheus-84bc4f9bcc-r9tvq

❯ ./target/release/linkerd2-cli-git-1ba566af-linux-amd64 jaeger list -A
Pods missing tracing configuration (restart these pods to enable tracing):
        * linkerd/linkerd-identity-7ccdfb455c-fbhsj
        * linkerd/linkerd-sp-validator-57878ff5cc-rshng
        * linkerd/linkerd-controller-74bbd789c7-59v75
        * linkerd/linkerd-destination-6f8977f44f-72bhg
        * linkerd/linkerd-proxy-injector-7664989ff9-gghrv
        * linkerd-viz/tap-759698d4c8-vrzmc
        * linkerd-viz/tap-injector-865cfdfb7d-89np8
        * linkerd-viz/metrics-api-9b9b4d79f-b42qp
        * linkerd-viz/grafana-5b8bc7dd5d-dtkqg
        * linkerd-viz/web-957c75976-8v7cx
        * linkerd-viz/prometheus-84bc4f9bcc-r9tvq
        * linkerd-jaeger/jaeger-69fc767648-ljvvl
        * linkerd-jaeger/jaeger-injector-7f4b5df5b4-n4lhr
        * linkerd-jaeger/collector-7f585dc68-7sdbh
```

**Comment by alpeb** [comment]
`linkerd jaeger check --proxy` now only checks the namespace exists. Maybe just make the whole thing no-op instead?

**Review by dadjeibaah — APPROVED** [review]
LGTM modulo other reviewers comments. 📦 🚢

**Pull request merged**